[Routing] Better inline requirements and defaults parsing

Foxprodev opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 4.4 
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40749 #40701
| License       | MIT
| Doc PR        | 

Remove `!` symbol from requirements and defaults array keys in Route class. Leave `!` symbol in Route compiled path for correct token creation.
**The only restriction I found:** 
Important variable can't get default value, only in UrlGenerator. 
As mentioned in 
https://github.com/symfony/symfony/blob/0f96ac74847d114c9d9679655bcf3e94b6ba69d1/src/Symfony/Component/Routing/RouteCompiler.php#L217
they are not optional

Feel free to help me with some advice. Thank you in advance

derrabus commented:
Thank you for your PR, @Foxprodev! I haven't reviewed your changes yet, but I noticed that you're about to submit a bugfix against the 5.x branch. Please note that we usually fix bugs on the **lowest maintained branch** that suffers from the bug. We currently maintain the branches 4.4 and 5.2, so if the bug is present there, please target one of those branches instead.

Foxprodev commented:
@derrabus Thanks, it makes sense. Well, I am going to test with 4.4 and rework branch according 4.4 version then :disappointed:

Commit:
[Route] Better inline requirements and defaults parsing

Remove ! symbol from requirements and defaults array keys in Route class. Leave ! symbol in Route compiled path for correct token creation
Added some inline route settings tests

Foxprodev reviewed (commented):
Done. No idea how to restart checks. Maybe an empty commit or amend?

nicolas-grekas commented:
Thank you @Foxprodev.
